unit-testing/presentation.html

```diff
@@ -1,27 +1,398 @@
+          jest.requireActual(moduleName)​​
+          jest.resetModules()​​
+          jest.isolateModules(fn)​​
+          // Mock modules​​
+          jest.fn(implementation?)​​
+          jest.spyOn(object, methodName)​​
+          jest.clearAllMocks()​​
+          jest.resetAllMocks()​​
+          jest.restoreAllMocks()​​
+        </code></pre>
+        </section>
+      </section>
+      <!-- 19 slide -->
+      <section>
+        <h3 class="slide-name">Snapshots​​</h3>
+        <ul>
+          <li class="fragment">Snapshot tests are used when you want to make sure your UI does not change unexpectedly.
+          </li>
+          <li class="fragment">A snapshot test renders a UI component by generating a serializable value for your React
+            tree -> takes a
+            snapshot -> compares it to the stored snapshot file.</li>
+          <li class="fragment">The test will fail if the two snapshots do not match.</li>
+          <li class="fragment">However snapshots themselves do not ensure that your component render logic is correct.
+          </li>
+          <li class="fragment">Use small snapshots because large snapshots can quickly become hard -> keep
+            no-large-snapshots rule.</li>
+        </ul>
+        <pre class="fragment"><code data-trim>
+          it('should match snapshot', () => {
+            const {toJSON} = render(<MyButton title="title"></MyButton>);
+            expect(toJSON()).toMatchSnapshot();
+          });</code></pre>
+      </section>
+      <!-- 20 slide -->
+      <section>
+        <!-- 20/0 slide -->
+        <section>
+          <h3 class="slide-name">Test coverage​</h3>
+          <ul>
+            <li class="fragment">With that being said it is generally accepted that 80% coverage is a good
+              goal to aim for.</li>
+            <li class="fragment">Trying to reach a higher coverage might turn out to be costly, while not
+              necessary producing enough benefit.​</li>
+            <li class="fragment">100% unit test coverage does not mean we had good tests, or even that the tests are
+              complete.</li>
+            <li class="fragment">The tests could be missing important data and only testing with data that succeeds,
+              failing to test data that causes failures.</li>
+            <li class="fragment">Merely executing the code without checking its functionality still counts in the
+              coverage metrics.</li>
+          </ul>
+        </section>
+        <!-- 20/1 slide -->
+        <section>
+          <img src='assets/images/coverage_1.png' height="500" />
+        </section>
+        <!-- 20/2 slide -->
+        <section>
+          <img src='assets/images/coverage_2.png' height="500" />
+        </section>
+        <!-- 20/3 slide -->
+        <section>
+          <img src='assets/images/coverage_3.png' height="500" />
+        </section>
+      </section>
+      <!-- 21 slide -->
+      <section>
```

Comment:
class="picture-slide", add header and remove image height attribute

Reply:
done1